pkgs/top-level/default.nix

```diff
@@ -19,83 +25,25 @@
                && system != "x86_64-solaris"
                && system != "x86_64-kfreebsd-gnu")
 
```

Comment:
I have a similar idea for `crossSystem`. Now that `crossStdenv` is properly staged, we can give each stdenv a `host` and `target` system+platform, and set them accordingly in the cross stages.

`crossStdenv` would remain as a entry-point argument, but `all-packages.nix` would not have that argument as just look at `stdenv.host` and `stdenv.target`.